fastjet/internal/delaunay/delaunay.go

```diff
@@ -473,3 +474,49 @@ func areCounterclockwise(a, b, c int) bool {
 	}
 	return a < b && c < a
 }
+
+// VoronoiCell returns the Vornoi points of a point in clockwise order
+// and the area those points enclose.
+//
+// If a point is on the border of the Delaunay triangulation the area will be Infinity
+// and the first and last point of the cell will be of a root triangle.
+func (d *Delaunay) VoronoiCell(p *Point) ([]*Point, float64) {
+	// border1 is set to the index of the first vornoi point that is part of a root triangle
+	border1 := -1
+	// border2 is set to the index of the first vornoi point that is part of a root triangle
+	border2 := -1
+	voronoi := make(points, len(p.adjacentTriangles))
+	t := p.adjacentTriangles[0]
+	// check whether the triangle contains any root points
+	if t.A.Equals(d.root.A) || t.A.Equals(d.root.B) || t.A.Equals(d.root.C) ||
+		t.B.Equals(d.root.A) || t.B.Equals(d.root.B) || t.B.Equals(d.root.C) ||
+		t.C.Equals(d.root.A) || t.C.Equals(d.root.B) || t.C.Equals(d.root.C) {
+		border1 = 0
+	}
+	x, y := t.circumcenter()
+	voronoi[0] = NewPoint(x, y)
+	for i := 1; i < len(p.adjacentTriangles); i++ {
+		t = p.findClockwiseTriangle(t)
+		// check whether the triangle contains any root points
+		if t.A.Equals(d.root.A) || t.A.Equals(d.root.B) || t.A.Equals(d.root.C) ||
+			t.B.Equals(d.root.A) || t.B.Equals(d.root.B) || t.B.Equals(d.root.C) ||
+			t.C.Equals(d.root.A) || t.C.Equals(d.root.B) || t.C.Equals(d.root.C) {
+			if border1 == -1 {
+				border1 = i
+			} else {
+				border2 = i
+			}
+		}
+		x, y = t.circumcenter()
+		voronoi[i] = NewPoint(x, y)
+	}
+	if border1 == -1 {
+		area := voronoi.polyArea()
+		return voronoi, area
+	}
+	if border1 != 0 || border2 != len(voronoi)-1 {
```

Comment:
I think this `if-check` could use some explanation.

at this point, `border1` is either `0` or `>= 1`.
but couldn't `border2` be `-1` or `> border1` ?

if `border2=-1`, we might have a problem with the slice-trick below.

there is also probably an issue when `len(voronoi) == 1` (or `len(p.adjacentTriangles) <= 1`)
(as `VoronoiCell` is an exported method, we should check for that pathological case. perhaps a bit around L488)

Reply:
done

Reply:
ok. perhaps document that the function will panic if the number of adjacent triangles is <= 3.